django-stubs/http/response.pyi

```diff
@@ -104,10 +104,13 @@ class HttpResponse(HttpResponseBase, Iterable[bytes]):
     def __iter__(self) -> Iterator[bytes]: ...
     def getvalue(self) -> bytes: ...
 
-class StreamingHttpResponse(HttpResponseBase, Iterable[bytes]):
-    streaming_content = _PropertyDescriptor[Iterable[object], Iterator[bytes]]()
-    def __init__(self, streaming_content: Iterable[object] = ..., *args: Any, **kwargs: Any) -> None: ...
+_I = TypeVar("_I", Iterable[object], AsyncIterable[object])
```

Comment:
Isn't it `_I = TypeVar("_I", bound=Union[Iterable[object], AsyncIterable[object]])`?

Reply:
[According to Django docs](https://docs.djangoproject.com/en/4.2/ref/request-response/#id6), `streaming_content` is "An iterator of the response content, bytestring encoded according to HttpResponse.charset."

Why use `object` here, shouldn't it be `Iterable[bytes], AsyncIterable[bytes]`?

Reply:
If you look at the code, Django will happily take anything that can be converted to bytes with the utility function: str, bytes and gettext string are all valid.

Reply:
> If you look at the code, Django will happily take anything that can be converted to bytes with the utility function: str, bytes and gettext string are all valid.

Sorry, I didn't realize that this loose behavior already existed before your changes, it was actually introduced in #712. So it's out of scope for this PR. Feel free to ignore my comment :)

But FWIW I don't agree with this approach. "It works" doesn't mean it's correct or intentional behavior, or it will continue to work with future Django versions. The point of type checkers is to introduce additional strictness to the otherwise fast-and-loose world of Python typing. I **want** type stubs to be stricter if that means it finds more bugs or leads to more precise code. Granted, we have to find a reasonable tradeoff between strictness and usefulness.

Reply:
Yes, you’re right, I mostly took what was already done for sync streaming responses. 

As mentioned in #712 , in this case, the Django code base is actually explicitly testing that strings are encoded (https://github.com/django/django/blob/6e32d1fa1dafd0c9cd9f93997ecebb26cd9a1b62/tests/httpwrappers/tests.py#L672), so I believe it’s more a question of having the documentation fixed. I’ll suggest the change. https://code.djangoproject.com/ticket/34556

Reply:
This was fixed in the doc.
https://docs.djangoproject.com/en/dev/ref/request-response/#django.http.StreamingHttpResponse

Reply:
So according to updated documentation it should be `bytes | str | StrPromise`, not `object`?

Reply:
And `| memoryview` as well then. As this would need to change for `HttpResponse` as well, I don’t know if this PR is the best place to do that, but I’ll follow your guideline.